### PR TITLE
[FIXED JENKINS-21485] AdministrativeMonitor for plugins that have failed

### PR DESCRIPTION
**Pull request by fbelzunc**
This PR substitute #2098 in case we keep @Vlatombe's PR #2172. 

Notice that I needed to do:

```
public final static PluginWrapperAdministrativeMonitor NOTICE = new PluginWrapperAdministrativeMonitor();
```

because it looks like the Extensions are not leaded at this point.

CC @jglick @daniel-beck @Vlatombe 

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/PluginWrapper.java [hunk 1]
```diff
@@ -567,13 +571,16 @@ public boolean hasLicensesXml() {
             PluginWrapper dependency = parent.getPlugin(d.shortName);
             if (dependency == null) {
                 missingDependencies.add(d.toString());
+                NOTICE.addErrorMessage(Messages.PluginWrapper_admonitor_MissingDependency(shortName, dependency.getLongName()));
```
Comment (Vlatombe) [inline core/src/main/java/hudson/PluginWrapper.java]: NPE here.

File: core/src/main/resources/hudson/Messages.properties [hunk 2]
```diff
@@ -70,4 +70,9 @@ ProxyConfiguration.FailedToConnectViaProxy=Failed to connect to {0}.
 ProxyConfiguration.FailedToConnect=Failed to connect to {0} (code {1}).
 ProxyConfiguration.Success=Success
 
-Functions.NoExceptionDetails=No Exception details
\ No newline at end of file
+Functions.NoExceptionDetails=No Exception details
+
+PluginWrapper.admonitor.OutdatedCoreVersion=Plugin {0} requires Jenkins {1} or later
+PluginWrapper.admonitor.MissingDependency=Plugin {0} doesn\u2019t have installed the required dependency {1}.
+PluginWrapper.admonitor.DisabledDependency=Plugin {0} has disabled the required dependency {1}
```
Comment (Vlatombe) [inline core/src/main/resources/hudson/Messages.properties]: > Plugin ssh-credentials has disabled the required dependency Credentials Plugin

I would rephrase as 'Plugin {0} depends on the disabled {1}.'

File: core/src/main/java/hudson/PluginWrapper.java [hunk 3]
```diff
@@ -555,6 +558,7 @@ public boolean hasLicensesXml() {
                 LOGGER.warning(shortName + " doesn't declare required core version.");
             } else {
                 if (Jenkins.getVersion().isOlderThan(new VersionNumber(requiredCoreVersion))) {
+                    NOTICE.addErrorMessage(Messages.PluginWrapper_admonitor_OutdatedCoreVersion(shortName, requiredCoreVersion));
```
Comment (Vlatombe) [inline core/src/main/java/hudson/PluginWrapper.java]: Use `getLongName()` instead of shortName for consistency?

File: core/src/main/resources/hudson/Messages.properties [hunk 4]
```diff
@@ -70,4 +70,9 @@ ProxyConfiguration.FailedToConnectViaProxy=Failed to connect to {0}.
 ProxyConfiguration.FailedToConnect=Failed to connect to {0} (code {1}).
 ProxyConfiguration.Success=Success
 
-Functions.NoExceptionDetails=No Exception details
\ No newline at end of file
+Functions.NoExceptionDetails=No Exception details
+
+PluginWrapper.admonitor.OutdatedCoreVersion=Plugin {0} requires Jenkins {1} or later
+PluginWrapper.admonitor.MissingDependency=Plugin {0} doesn\u2019t have installed the required dependency {1}.
```
Comment (Vlatombe) [inline core/src/main/resources/hudson/Messages.properties]: > Plugin ssh-slaves doesn’t have installed the required dependency ssh-credentials.

I suggest `Plugin {0} requires the missing plugin {1}.` instead.

**Comment by Vlatombe** [comment]
Should we restore #2172 in this PR as well since it got reverted in 2.0?

**Comment by oleg-nenashev** [comment]
Not ready to merge. @fbelzunc Are you working on it?

**Comment by daniel-beck** [comment]
@Vlatombe @fbelzunc I'd love to see #2172 and this revived soon, so we make sure plugin loading is as solid and errors as noticeable and understandable as we can make it (JENKINS-34073).

**Comment by daniel-beck** [comment]
Superseded by #2487.
